test/image/mocks/bar_text_relative.json

```diff
@@ -0,0 +1,40 @@
+{
+    "data":[
+        {
+            "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3,4],
+	    "textposition":"outside",
+            "y":[1,2,3,4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"This bar is size one",0],
+	    "textposition":"auto",
+            "y":[3,2,1,0],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":1,
+            "text":[-1,-3,-2,-4],
+	    "textposition":"inside",
+            "y":[-1,-3,-2,-4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "text":[0,"very small size",-3,-2],
```

Comment:
@alexcjohnson I've just realised that my implementation for `outside` labels doesn't do what I said :disappointed: . How do you think it should be? Shall I rotate it like in the case of `inside` labels?

---

:+1: Yes, I'll push the labels close to the bar ends. One question: at the moment I'm using a 1px padding, shall I increase it (the padding in [the ggplot2 example](http://stackoverflow.com/questions/6644997/showing-data-values-on-stacked-bar-chart-in-ggplot2) looks too much, though)?